test/test_other.py

```diff
@@ -12426,7 +12332,7 @@ def test_main_reads_params(self):
 
   def test_INCOMING_MODULE_JS_API(self):
     def test(args):
-      self.run_process([EMCC, test_file('hello_world.c'), '-O3', '--closure=1', '-sENVIRONMENT=node,shell', '--output-eol=linux'] + args)
+      self.run_process([EMCC, test_file('hello_world.c'), '-O3', '--closure=1', '-sENVIRONMENT=web', '--output-eol=linux'] + args)
```

Comment:
Maybe make this change separately since I imagine it will effect the code size numbers and it would be nice if this PR did not change them.

Reply:
So this change was ported over from the removed `test_small_js_flags` in `test_browser.py`.

These tests were identical, except for `ENVIRONMENT`, so it made sense to merge them into one, and usually we care about code sizes for `ENVIRONMENT=web` and not shells anyway so I chose to keep this option when merging.

Reply:
I see... Are you sure the browser test is no longer useful?  One difference is that the browser test actually runs the resulting program, but maybe that is not important here?

Reply:
> but maybe that is not important here?

Yeah, exactly - this test is literally hello world and we run it (and much more complex tests) plenty of times throughout the test suite. The only thing it was used for is code size snapshot, but then we don't need to run it in the browser to get that.

Reply:
I reverted this test merge to keep it simple. The browser test is still moved to test_other.py, but stays separate.